tests/test_docs/test_guides.py

```diff
@@ -0,0 +1,194 @@
+
+        txt_files = sorted(list(self.snippet_path.rglob("*.txt")))
+
+        self.commands = []
+
+        for txt_file in txt_files:
+            with open(txt_file, "r", encoding="utf-8") as fp:
+                self.commands += [line.strip() for line in fp.readlines()]
+
+    def run_commands(self, commands: Optional[List[str]] = None) -> None:
+        """Run commands"""
+
+        print(f"\nSet execution path to in {self.execution_path}")
+        os.chdir(self.execution_path)
+
+        if not commands:
+            commands = self.commands
+
+        # Command loop
+        for cmd in commands:
+            try:
+                print(f"> {cmd}")
+
+                # Replace placeholders
+                for old_value, new_value in self.placeholders.items():
+                    cmd = cmd.replace(old_value, new_value)
+
+                # Run pre-callbacks
+                pre_callback = self.pre_callbacks.get(cmd, None)
+                if pre_callback:
+                    print(f">  Running pre-callback: {pre_callback['name']}")
+                    pre_callback["callback"]()  # type: ignore
+
+                # Treat cd commands separately
+                if cmd.startswith("cd "):
+                    os.chdir(cmd[3:])
+                    continue
+
+                # Treat pipes separately
+                if "|" in cmd:
+                    cmds = [i.strip() for i in cmd.split("|")]
+                    assert len(cmds) == 2, "Only piping two commands is supported"
+                    process = subprocess.Popen(  # nosec
+                        cmds[0].split(" "), stdout=subprocess.PIPE
+                    )
+                    subprocess.check_output(  # nosec
+                        cmds[1].split(" "), stdin=process.stdout
+                    )
+                    process.wait()
+                    assert (
+                        process.returncode == 0
+                    ), f"Guide execution has failed at command '{cmd}'\n{process.returncode}"
+                    continue
+
+                # Run command
+                process = subprocess.run(cmd.split(" "), timeout=DEFAULT_TIMEOUT)  # type: ignore  # nosec
+                assert (
+                    process.returncode == 0
+                ), f"Guide execution has failed at command '{cmd}'\n{process.returncode}"
+
+            except subprocess.TimeoutExpired:
+                # We expect some commands to never end (for example running a service)
+                # so we allow for timeouts
+                continue
+
+            except Exception as e:
+                raise AssertionError(
+                    f"Guide execution has failed at command '{cmd}'\n{e}"
+                )
+
+    def teardown(self) -> None:
+        """Teardown"""
+        # Remove execution directory
+        remove_test_directory(str(self.execution_path))
+
+        # Run teardown commands
+        self.run_commands(self.teardown_commands)
+
+
+@pytest.mark.e2e
```

Comment:
I think the test is not running. I've marked it e2e but I don't see it in the logs. Maybe the e2e jobs only runs tests from a specific folder, need to investigate. It passes locally.